compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEUtils.cpp

```diff
@@ -298,34 +287,44 @@ bool isMatmul(linalg::LinalgOp linalgOp) {
          match6DLinalgGenericMatmul(linalgOp);
 }
 
+/// Utility to identify if the input operand has matmul-like op in its
+/// def-chain.
+bool isMatmulInDefChain(Value operand) {
+  if (isa<BlockArgument>(operand)) return false;
+  while (Operation *defOp = operand.getDefiningOp()) {
```

Comment:
I guess you can completely eliminate the `if` check here. The `while` check that follows would anyway help in that.